src/screens/WasteCollectionScreen.js

```diff
@@ -0,0 +1,299 @@
+import Autocomplete from 'react-native-autocomplete-input';
+import { Calendar, LocaleConfig } from 'react-native-calendars';
+import { ScrollView, TouchableOpacity } from 'react-native';
+
+import {
+  Icon,
+  RegularText,
+  SafeAreaViewFlex,
+  WrapperWithOrientation,
+  WasteCalendarLegend,
+  Wrapper,
+  HeaderLeft,
+  LoadingContainer
+} from '../components';
+import { colors, device, normalize, texts } from '../config';
+import { arrowLeft, arrowRight } from '../icons';
+import { useQuery } from 'react-apollo';
+import { getQuery, QUERY_TYPES } from '../queries';
+import { useRefreshTime } from '../hooks';
+import { graphqlFetchPolicy } from '../helpers';
+import { NetworkContext } from '../NetworkProvider';
+
+LocaleConfig.locales['de'] = {
+  monthNames: [
+    'Januar',
+    'Februar',
+    'März',
+    'April',
+    'Mai',
+    'Juni',
+    'Juli',
+    'August',
+    'September',
+    'Oktober',
+    'November',
+    'Dezember'
+  ],
+  monthNamesShort: [
+    'Jan.',
+    'Feb.',
+    'Mär.',
+    'Apr.',
+    'Mai',
+    'Jun.',
+    'Jul.',
+    'Aug.',
+    'Sep.',
+    'Okt.',
+    'Nov.',
+    'Dez.'
+  ],
+  dayNames: ['Sonntag', 'Montag', 'Dienstag', 'Mittwoch', 'Donnerstag', 'Freitag', 'Samstag'],
+  dayNamesShort: ['So.', 'Mo.', 'Di.', 'Mi.', 'Do.', 'Fr.', 'Sa.'],
+  today: 'Heute'
+};
+LocaleConfig.defaultLocale = 'de';
+
+const getMarkedDates = (types, streetData) => {
+  let markedDates = {};
+
+  const wasteLocationTypes = streetData?.[QUERY_TYPES.WASTE_ADDRESSES]?.[0]?.wasteLocationTypes;
+
+  if (wasteLocationTypes && types) {
+    wasteLocationTypes.forEach((wasteLocationType) => {
+      // only add marked dates for known types
+      if (!types[wasteLocationType.wasteType]) {
+        return;
+      }
+
+      const { color, selected_color: selectedColor } = types[wasteLocationType.wasteType];
+      wasteLocationType?.listPickUpDates?.forEach((date) => {
+        markedDates[date] = {
+          dots: [...(markedDates[date]?.dots ?? []), { color, selectedColor }]
+        };
+      });
+    });
+  }
+
+  const today = moment().format('YYYY-MM-DD');
+
+  // highlight today
+  markedDates[today] = {
+    ...(markedDates[today] ?? {}),
+    selected: true,
+    selectedColor: colors.lighterPrimary
+  };
+
+  return markedDates;
+};
+
+const renderArrow = (direction) =>
+  direction === 'right' ? (
+    <Icon xml={arrowRight(colors.primary)} style={styles.icon} />
+  ) : (
+    <Icon xml={arrowLeft(colors.primary)} style={styles.icon} />
+  );
+
+// show streets that contain the string in it
+// return empty list on an exact match (except for capitalization)
+const filterStreets = (currentInputValue, streetData) => {
+  if (currentInputValue === '') return [];
+
+  return streetData
+    ?.filter((street) =>
+      getStreetString(street).toLowerCase().includes(currentInputValue.toLowerCase())
+    )
+    .filter((street) => getStreetString(street) !== currentInputValue)
+    .slice(0, 5);
```

Comment:
`slice` to always return a maximum of five results?

Reply:
yes. this is simply to limit the number of suggestions